src/app/helpers/custom-errors/baw-api-error.ts

```diff
@@ -0,0 +1,78 @@
+export class BawApiError extends Error {
+  public name = "BawApiError";
+  public status: number;
+  public info: Record<string, string>;
+  private _message: string;
+
+  public constructor(error: ApiErrorDetails);
+  public constructor(
+    status: number,
+    message: string,
+    info?: Record<string, string>
+  );
+
+  public constructor(...args: any[]) {
+    super(args.length === 1 ? args[0].message : args[1]);
+
+    this.status = args.length === 1 ? args[0].status : args[0];
+    this._message = args.length === 1 ? args[0].message : args[1];
+    this.info = args.length === 1 ? args[0].info : args[2] ?? {};
```

Comment:
this is pretty messy IMO

as far as i can tell we have a maximum of three arguments... no need for variadic args. 

Just name the args? Or can we simplify the call sites?